### PR TITLE
Implemented Security block for Tutor profile edit

### PR DESCRIPTION
**Pull request by abalanovsky**
Implemented Security block for Tutor profile edit

Знімок екрана 2024-03-27 о 18 47 25

Знімок екрана 2024-03-27 о 18 47 49

### CODE REVIEW

File: src/components/confirm-dialog/ConfirmDialog.tsx [hunk 1]
```diff
@@ -48,12 +50,25 @@ const ConfirmDialog: FC<ConfirmDialogProps> = ({
...
       <DialogActions sx={styles.actions}>
-        <AppButton onClick={onConfirm}>
-          {confirmButton ? confirmButton : t('common.yes')}
-        </AppButton>
-        <AppButton onClick={onDismiss} variant={ButtonVariantEnum.Tonal}>
-          {cancelButton ? cancelButton : t('common.no')}
-        </AppButton>
+        {revertButtons ? (
+          <>
+            <AppButton onClick={onDismiss} variant={ButtonVariantEnum.Tonal}>
```
Comment (Tolik170) [inline src/components/confirm-dialog/ConfirmDialog.tsx:55]: As I see, the buttons are the same, you are just changing their places, try to use for example flexDirection property for reversing them, and pass revertButton to styles for applying appropriate direction

File: src/containers/tutor-profile/security-block/SecurityBlock.styles.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      mb: '30px'
+    }
+  },
+  subtitle: {
+    typography: TypographyVariantEnum.Body1,
+    mb: '15px'
+  },
+  passwordButtonsContainer: {
+    mt: '10px',
+    mb: '20px'
```
Comment (Tolik170) [inline src/containers/tutor-profile/security-block/SecurityBlock.styles.js:32]: just use a single m property

File: src/containers/tutor-profile/security-block/SecurityBlock.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+import AppButton from '@mui/material/Button'
+import Divider from '@mui/material/Divider'
+import { t } from 'i18next'
+
+import AppTextField from '~/components/app-text-field/AppTextField'
+import ConfirmDialog from '~/components/confirm-dialog/ConfirmDialog'
+import TitleWithDescription from '~/components/title-with-description/TitleWithDescription'
+import useForm from '~/hooks/use-form'
+
+import { styles } from './SecurityBlock.styles'
```
Comment (dmtrth25) [inline src/containers/tutor-profile/security-block/SecurityBlock.tsx:13]: forgot `aliases`

File: src/components/confirm-dialog/ConfirmDialog.tsx [hunk 4]
```diff
@@ -48,12 +50,25 @@ const ConfirmDialog: FC<ConfirmDialogProps> = ({
...
+              {cancelButton ? cancelButton : t('common.no')}
```
[the remark this answers is not in the document]
Comment (abalanovsky) [reply src/components/confirm-dialog/ConfirmDialog.tsx:56]: There will be so much conditionals, i'll refactor the code according to Anatoly's suggestion

[the remark this answers is not in the document]
Comment (Tolik170) [reply src/components/confirm-dialog/ConfirmDialog.tsx:56]: I guess you can do it like:
`{ cancelButton ?? t('common.no')}`
same for confirmButton

**Review by dmtrth25 — APPROVED** [review]

File: src/containers/tutor-profile/security-block/SecurityBlock.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+          type={'password'}
+          value={data.password}
+        />
+        <AppTextField
+          errorMsg={t(errors.confirmPassword)}
+          fullWidth
+          label={t('editTutor.passwordSecurityTab.retypePassword')}
+          onBlur={handleBlur('confirmPassword')}
+          onChange={handleInputChange('confirmPassword')}
+          type={'password'}
```
Comment (Tolik170) [inline src/containers/tutor-profile/security-block/SecurityBlock.tsx:69]: add password to InputEnum

**Review by Tolik170 — APPROVED** [review]

**Review by Mav-Ivan — APPROVED** [review]

**Pull request merged**